docs/source/conf.py

```diff
@@ -441,15 +453,289 @@ def check_timeout_and_api_kwargs_presence(obj: object) -> int:
+    # i.e. {Message: ".. admonition: Returned in ..."}
+    admonition_for_class_name = {}
+
+    for cls, methods in methods_for_class_name.items():
+        admonition = """
+
+.. admonition:: Returned in
+    :class: returned-in
+"""
+        if len(methods) > 1:
+            for method_name in sorted(methods):
+                admonition += "\n    * " + f":meth:`telegram.Bot.{method_name}`"
+        else:
+            admonition += f"\n    :meth:`telegram.Bot.{methods[0]}`"
+
+        admonition += "\n    "  # otherwise an unexpected unindent warning will be issued
+        admonition_for_class_name[cls] = admonition
+
+    return admonition_for_class_name
+
+
+RETURN_ADMONITION_FOR_CLASS_NAME = create_return_admonitions()
+
+
+def create_use_in_admonitions() -> dict[str, str]:
+    """Creates 'Use in' admonitions for classes whose instances that are accepted as arguments for
+    Bot's methods.
+    """
+
+    # First, generate a mapping of class names to the methods which accept them as arguments,
+    # i.e. {InlineQueryResult: [answer_inline_query, ...]}
+    methods_for_class_name = defaultdict(set)  # using set here because there can be repetitions
+
+    forward_ref_pattern = re.compile(r"ForwardRef\('(?P<class_name>.+?)'\)")  # non-greedy
+    telegram_pattern = re.compile(r"telegram\.(?P<class_name>[a-zA-Z_.]+)")
+    skip_pattern = re.compile(
+        r"^typing.Union\[ForwardRef\('DefaultValue\[DVType]'\), ((float)|(bool)|(int)|(str))"
+        r"(, NoneType)?]$"
+    )
+
+    for method in [
+        m[0]
+        for m in inspect.getmembers(telegram.Bot, inspect.isfunction)  # not .ismethod
+        if not m[0].startswith("_")
+        and m[0].islower()  # islower() to avoid camelCase methods
+        and m[0] in telegram.Bot.__dict__  # method is not inherited from TelegramObject
+    ]:
+
+        sig = inspect.signature(getattr(telegram.Bot, method))
+        parameters = sig.parameters
+
+        for type_annotation in parameters.values():
+            annotation = type_annotation.annotation
+            annotation_text = str(annotation)
+
+            if skip_pattern.match(annotation_text):
+                continue
+
+            if isinstance(annotation, type) and "telegram" in annotation_text:
+                methods_for_class_name[str(annotation)].add(method)
+
+            # this is a _UnionGenericAlias
+            elif "telegram" in annotation_text or "ForwardRef" in annotation_text:
+                matches = [m for m in forward_ref_pattern.finditer(annotation_text)] + [
+                    m for m in telegram_pattern.finditer(annotation_text)
+                ]
+
+                for match in matches:
+                    # resolving class
+                    try:
+                        klass = eval(f"telegram.{match.group('class_name')}")
+                    except AttributeError:
+                        klass = eval(f"telegram.ext.{match.group('class_name')}")
```

Comment:
does this handle cases like `telegram.ext.filters.Sticker` as well?

Reply:
Well it didn't, I just added this functionality. In practice, though, docstrings with `filters` use full class name. I doubt you will put `:class: Sticker` if you mean `filters.Sticker`. But this addition is reasonable anyway (I could imagine someone putting `:class: BaseFilter` into the docstring).